File: bench/ml/37_range_builtin.ml
# range() builtin: MyLang range() eagerly builds a whole array (like Python 2).
# The Python version uses list(range(N)) to match that allocation; idiomatic
# Python 3 would use the lazy range object (faster) - see the README note.
var scale = 1;
if (len(argv) > 0)
    scale = int(argv[0]);

var N = 1000000 * scale;
var s = 0;

foreach (var e in range(N))
    s += e;

print("result:", s);

File: bench/py/37_range_builtin.py
# list(range(N)) matches MyLang's eager range() allocation (not lazy range).
import sys
scale = int(sys.argv[1]) if len(sys.argv) > 1 else 1

N = 1000000 * scale
s = 0

for e in list(range(N)):
    s += e

print("result:", s)
